src/Block/BlockContextManager.php

```diff
@@ -134,6 +134,8 @@ public function get($meta, array $settings = []): BlockContextInterface
                 // merge user settings
                 $settings = array_merge($meta['settings'], $settings);
             }
+
+            $block->setSettings($settings);
```

Comment:
Are we supposed to also set the Settings in the `else` part ?

Reply:
Not sure about this. When you pass a `BlockInterface` in the else part your block could already have valid settings.

Reply:
We doing later
```
try {
            $settings = $this->resolve($block, array_merge($block->getSettings(), $settings));
        } catch (ExceptionInterface $e) {
```
So this means that the $settings might not be compatible with the Block provided.

That's why I wonder if we shouldn't set the settings after the try/catch call